bundles/org.openhab.binding.wemo/src/main/java/org/openhab/binding/wemo/internal/handler/WemoLightHandler.java

```diff
@@ -124,28 +118,30 @@ public void initialize() {
     public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
         if (bridgeStatusInfo.getStatus().equals(ThingStatus.ONLINE)) {
             updateStatus(ThingStatus.ONLINE);
-            onSubscription();
-            onUpdate();
         } else {
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.BRIDGE_OFFLINE);
-            ScheduledFuture<?> job = refreshJob;
+            ScheduledFuture<?> job = this.pollingJob;
             if (job != null && !job.isCancelled()) {
                 job.cancel(true);
             }
-            refreshJob = null;
+            this.pollingJob = null;
         }
     }
 
     @Override
     public void dispose() {
         logger.debug("WeMoLightHandler disposed.");
 
-        ScheduledFuture<?> job = refreshJob;
+        ScheduledFuture<?> job = this.pollingJob;
         if (job != null && !job.isCancelled()) {
             job.cancel(true);
         }
-        refreshJob = null;
+        this.pollingJob = null;
         removeSubscription();
+        UpnpIOService localService = service;
+        if (localService != null) {
+            localService.unregisterParticipant(this);
+        }
```

Comment:
Why? This is already done by `removeSubscription()` called from the line above?

Reply:
done